src/groups/mqb/mqbauthn/mqbauthn_anonauthenticator.cpp

```diff
@@ -0,0 +1,209 @@
+// Copyright 2016-2025 Bloomberg Finance L.P.
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <mqbauthn_anonauthenticator.h>
+
+// MQB
+#include <mqbcfg_messages.h>
+#include <mqbplug_authenticator.h>
+
+// BDE
+#include <bsl_iostream.h>
+#include <bsl_memory.h>
+#include <bsl_optional.h>
+#include <bsl_string.h>
+#include <bsl_string_view.h>
+#include <bsla_annotations.h>
+#include <bslma_allocator.h>
+#include <bslma_managedptr.h>
+#include <bsls_assert.h>
+
+namespace BloombergLP {
+namespace mqbauthn {
+
+// ------------------------------
+// class AnonAuthenticationResult
+// ------------------------------
+
+AnonAuthenticationResult::AnonAuthenticationResult(
+    bsl::string_view                  principal,
+    bsl::optional<bsls::Types::Int64> lifetimeMs,
+    bslma::Allocator*                 allocator)
+: d_principal(principal, allocator)
+, d_lifetimeMs(lifetimeMs)
+{
+}
+
+AnonAuthenticationResult::~AnonAuthenticationResult()
+{
+}
+
+bsl::string_view AnonAuthenticationResult::principal() const
+{
+    return d_principal;
+}
+
+const bsl::optional<bsls::Types::Int64>&
+AnonAuthenticationResult::lifetimeMs() const
+{
+    return d_lifetimeMs;
+}
+
+// -----------------------
+// class AnonAuthenticator
+// -----------------------
+
+AnonAuthenticator::AnonAuthenticator(
+    const mqbcfg::AuthenticatorPluginConfig* config,
+    bslma::Allocator*                        allocator)
+: d_allocator_p(allocator)
+, d_isStarted(false)
+, d_shouldPass(true)
+{
+    if (!config) {
+        BALL_LOG_INFO
+            << "No configuration provided, configuring AnonAuthenticator with "
+               "the default shouldPass = true";
+        return;
+    }
+
+    // Load the configured `shouldPass` argument
+    bool isShouldPassFound = false;
+    bsl::vector<mqbcfg::PluginSettingKeyValue>::const_iterator it =
+        config->settings().cbegin();
+    for (; it != config->settings().cend(); ++it) {
+        if (it->key() == "shouldPass") {
+            if (!it->value().isBoolValValue()) {
+                BALL_LOG_WARN << "Expected bool for 'shouldPass' setting...";
```

Comment:
```suggestion
                BALL_LOG_WARN << "Expected bool for 'shouldPass' setting, got type id = " << it->value().selectionId();
```
Possible to add more verbosity here and print type index. It might help to debug one day if this warning is triggered